### PR TITLE
Run tests on multiple os

### PR DESCRIPTION
**Pull request by wyzlle**
Run MSRV, generate plan & library tests on ubuntu, windows and macos

### CODE REVIEW

**Review by Milo123459 — APPROVED** [review]

**Comment by nebulatgs** [comment]
Can we also run the MSRV tests on a matrix?

**Review by nebulatgs — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -102,6 +105,7 @@ jobs:
           args: --package nixpacks --lib --test generate_plan_tests
       - name: List docker tests (prep test-docker)
         id: docker-prep
+        if: matrix.os == 'ubuntu-latest'
```
Comment (Milo123459) [inline .github/workflows/ci.yml:108]: Is there a reason we only run docker tests on ubuntu? Just curious

Comment (nebulatgs) [reply .github/workflows/ci.yml:108]: yeah they break on windows bc no buildkit

**Review by Milo123459 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Milo123459) [reply .github/workflows/ci.yml:108]: What about macos?

**Review by coffee-cup — APPROVED** [review]

**Pull request merged**